src/libraries/System.Private.CoreLib/src/System/Environment.Win32.cs

```diff
@@ -220,6 +220,13 @@ private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOptio
 
             switch (folder)
             {
+                // Special-cased values to not use SHGetFolderPath when we have a more direct option available.
+                case SpecialFolder.System when option == SpecialFolderOption.None:
```

Comment:
This means that `GetFolderPath(SpecialFolder.System, SpecialFolderOption.None)` on Windows Nano is going to return the system directory, but `GetFolderPath(SpecialFolder.System, SpecialFolderOption.Create)` or `GetFolderPath(SpecialFolder.System, SpecialFolderOption.DoNotVerify)` is going to return empty string.

I think we should return the SystemDirectory unconditionally here.

Reply:
Makes sense. Can we assume the system directory always exists?  Presumably yes?

Reply:
I think so.